types/react-select/lib/stateManager.d.ts

```diff
@@ -1,25 +1,28 @@
 import { Component, ComponentType, Ref as ElementRef } from 'react';
 
+import SelectBase, { Props as SelectProps } from './Select';
 import { ActionMeta, InputActionMeta, ValueType } from './types';
 
-export interface Props<OptionType> {
+export interface DefaultProps<OptionType> {
   defaultInputValue?: string;
   defaultMenuIsOpen?: boolean;
   defaultValue?: ValueType<OptionType>;
-  inputValue?: string;
-  menuIsOpen?: boolean;
-  value?: ValueType<OptionType>;
 }
 interface State<OptionType> {
   inputValue: string;
   menuIsOpen: boolean;
   value: ValueType<OptionType>;
 }
 
-export class StateManager<OptionType> extends Component<Props<OptionType>, State<OptionType>> {
-  static defaultProps: Props<any>;
+type GetOptionType<T> = T extends SelectBase<infer OT> ? OT : never;
 
-  select: ElementRef<any>;
+export class StateManager<
+  OptionType = { label: string; value: string },
+  T extends SelectBase<OptionType> = SelectBase<OptionType>
+> extends Component<SelectProps<OptionType> & DefaultProps<OptionType>, State<OptionType>> {
```

Comment:
Looking at the original sources, I would recommend to reference `Props` and add the type `StateProps`.

https://github.com/JedWatson/react-select/blob/master/src/stateManager.js#L53

https://github.com/JedWatson/react-select/blob/master/src/stateManager.js#L25-L36

Reply:
Done as well.